### PR TITLE
Add lifecycle plugin

### PR DESCRIPTION
**Pull request by HidenoriMatsubayashi**
I added the lifecycle plugin. But it doesn't be called from anywhere now because we need to consider the specification of the app lifecycle. Also, I added `DEBUG` log level into the logger.

### CODE REVIEW

**Comment by HidenoriMatsubayashi** [comment]
@taka-okayama Can you review the PR?

File: src/flutter/shell/platform/linux_embedded/plugin/lifecycle_plugin.h [hunk 1]
```diff
@@ -0,0 +1,34 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
```
Comment (taka-okayama) [inline src/flutter/shell/platform/linux_embedded/plugin/lifecycle_plugin.h:1]: Is this correct that this copyright is different from the one in lifecycle_plugin.cc?

Comment (HidenoriMatsubayashi) [reply src/flutter/shell/platform/linux_embedded/plugin/lifecycle_plugin.h]: It's copypasta. I fixed now.

Comment (taka-okayama) [reply src/flutter/shell/platform/linux_embedded/plugin/lifecycle_plugin.h]: thx

**Review by taka-okayama — APPROVED** [review]

**Pull request merged**